src/main/java/DilemmaDetector/Simulator/CollisionDetector.java

```diff
@@ -25,16 +25,18 @@ public CollisionDetector(Model model, Actor mainVehicle,
     public List<Actor> detectCollisionInMoment() {
         List<Actor> collidedActors = new LinkedList<>();
         for (RigidBody surroundingRigidBody: surroundingRigidBodies) {
-            if (detectCollisionWithSurrounding(surroundingRigidBody)) {
+            if (detectCollisionWithRigidBodyInMoment(surroundingRigidBody, "surrounding")) {
```

Comment:
Może surrounding może składać się z aktorów? Takich, którzy się nie przesuwają. Sugestia do rozważenia, ale odpowiedz tutaj czy chcesz aby tak było czy nie, to Ty implementujesz. Wpadło mi to do głowy, bo Aktor składa się z rigidBody i Stringa :P

Reply:
Tak właśnie myśłałem ale aktor się kojarzył z ruchem

Reply:
Mogę tak zrobić to w sumie dużo roboty nie powinno być